pennylane/decomposition/symbolic_decomposition.py

```diff
@@ -16,237 +16,158 @@
+    """Decompose the power of a self-adjoint operator, assumes z is an integer."""
 
-@register_resources(_pow_resource)
-def repeat_pow_base(*_, base, z, **__):
-    """Decompose the power of a gate."""
-    assert isinstance(z, int) and z >= 0
-    for _ in range(z):
-        base._unflatten(*base._flatten())  # pylint: disable=protected-access
-
-
-def _pow_pow_resource(base_class, base_params, z):  # pylint: disable=unused-argument
-    """Resources of the power of the power of a gate."""
-    base_class, base_params, base_z = (
-        base_params["base_class"],
-        base_params["base_params"],
-        base_params["z"],
-    )
-    return {pow_resource_rep(base_class, base_params, z * base_z): 1}
-
-
-@register_resources(_pow_pow_resource)
-def merge_powers(*_, base, z, **__):
-    """Decompose the power of the power of a gate."""
-    qml.pow(base.base, z=z * base.z)
-
-
-@functools.lru_cache(maxsize=1)
-def same_type_adjoint_ops():
-    """A set of operators whose adjoint is an instance of its own type."""
-    return frozenset(
-        {
-            # identity
-            qml.Identity,
-            qml.GlobalPhase,
-            # non-parametric gates
-            qml.H,
-            qml.X,
-            qml.Y,
-            qml.Z,
-            qml.SWAP,
-            qml.ECR,
-            # single-qubit parametric gates
-            qml.Rot,
-            qml.U1,
-            qml.U2,
-            qml.U3,
-            qml.RX,
-            qml.RY,
-            qml.RZ,
-            qml.PhaseShift,
-            # multi-qubit parametric gates
-            qml.MultiRZ,
-            qml.PauliRot,
-            qml.PCPhase,
-            qml.IsingXX,
-            qml.IsingYY,
-            qml.IsingZZ,
-            qml.IsingXY,
-            qml.PSWAP,
-            qml.CPhaseShift00,
-            qml.CPhaseShift01,
-            qml.CPhaseShift10,
-            # matrix gates
-            qml.QubitUnitary,
-            qml.DiagonalQubitUnitary,
-            qml.BlockEncode,
-            qml.SpecialUnitary,
-            # custom controlled ops
-            qml.CH,
-            qml.CY,
-            qml.CZ,
-            qml.CNOT,
-            qml.CSWAP,
-            qml.CCZ,
-            qml.Toffoli,
-            qml.MultiControlledX,
-            qml.CRX,
-            qml.CRY,
-            qml.CRZ,
-            qml.CRot,
-            qml.ControlledPhaseShift,
-            # arithmetic ops
-            qml.QubitSum,
-            qml.IntegerComparator,
-            # qchem ops
-            qml.SingleExcitation,
-            qml.SingleExcitationMinus,
-            qml.SingleExcitationPlus,
-            qml.DoubleExcitation,
-            qml.DoubleExcitationPlus,
-            qml.DoubleExcitationMinus,
-            qml.OrbitalRotation,
-            qml.FermionicSWAP,
-            # templates
-            qml.CommutingEvolution,
-        }
-    )
+    def f():
+        base._unflatten(*base._flatten())
+
+    qml.cond(z % 2 == 1, f)()
```

Comment:
I'm bit confused here: do we assume all the operators that we are to take care of using decomp are unitary by default? Since without a guarantee of unitary, self-adjoint operators are not necessary involutory (Op^2 == 1)

Reply:
I remember there used to be some other on-going effort to merge Hermitian observable (basically the same as self-adjoint operator) into operator. If this part of new operators are also gonna be interacting with decomposition system then I anticipate some hardship in the future both conceptually and technically

Reply:
I think in the context of decompositions this should be true. Decompositions are applied on operations specifically and not observables in general.

Reply:
Renamed to `pow_involutory`